include/nil/crypto3/marshalling/algebra/types/detail/curve_element/basic_type.hpp

```diff
@@ -101,36 +99,42 @@ namespace nil {
 
                         template<typename TIter>
                         nil::marshalling::status_type read(TIter &iter, std::size_t size) {
+                            using reader = crypto3::marshalling::processing::curve_element_reader<
+                                bit_length(), typename base_impl_type::endian_type, CurveGroupType>;
                             // if (size < length()) {
                             //     return nil::marshalling::status_type::not_enough_data;
                             // }
 
-                            read_no_status(iter);
+                            auto status = reader::process(value(), iter);
```

Comment:
Wouldn't it be better to directly specify type as 
`nil::marshalling::status_type`
instead of `auto` inference?